packages/rich-text/src/plugins/Hr/index.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import * as Slate from 'slate-react';
+import { css, cx } from 'emotion';
+import tokens from '@contentful/forma-36-tokens';
+import { EditorToolbarButton } from '@contentful/forma-36-react-components';
+import { Transforms, Editor } from 'slate';
+import { BLOCKS } from '@contentful/rich-text-types';
+import { useCustomEditor } from '../../hooks/useCustomEditor';
+import { CustomElement, CustomEditor } from 'types';
+
+const styles = {
+  container: css`
+    margin: 0 0 ${tokens.spacingL};
+  `,
+  hr: css`
+    margin: 0;
+    height: ${tokens.spacingM};
+    background: transparent;
+    position: relative;
+    border: 0;
+    user-select: none;
+    &:hover {
+      cursor: pointer;
+    }
+    &::after {
+      content: '';
+      position: absolute;
+      width: 100%;
+      height: 1px;
+      background: ${tokens.colorElementMid};
+      top: 50%;
+    }
+  `,
+  hrSelected: css`
+    &::after {
+      background: ${tokens.colorPrimary};
+      -webkit-box-shadow: 0px 0px 5px ${tokens.colorPrimary};
+      box-shadow: 0px 0px 5px ${tokens.colorPrimary};
+    }
+  `,
+};
+
+interface ToolbarHrButtonProps {
+  isDisabled?: boolean;
+}
+
+export function withHrEvents(editor: CustomEditor, event: KeyboardEvent) {
+  if (!editor.selection) return;
+
+  const [currentFragment] = Editor.fragment(editor, editor.selection.focus.path) as CustomElement[];
+  const isEnter = event.keyCode === 13;
+  const isCurrentBlockHr = currentFragment.type === BLOCKS.HR;
+
+  if (isEnter && isCurrentBlockHr) {
+    event.preventDefault();
+
+    editor.moveToTheNextLine();
+  }
+}
+
+export function ToolbarHrButton(props: ToolbarHrButtonProps) {
+  const editor = useCustomEditor();
+
+  function handleOnClick() {
+    const hr = {
+      type: BLOCKS.HR,
+      children: [{ text: '' }],
+      isVoid: true,
```

Comment:
I don't know what slate expects, but it seems odd that this void block would have a text child 🤔

Reply:
It's how it's supposed to be from now on.. I think they have it explained on their documentation.